src/Execution/Arguments/UpsertModel.php

```diff
@@ -22,20 +26,39 @@ public function __construct(callable $previous)
     public function __invoke($model, $args): mixed
     {
         // TODO consider Laravel native ->upsert(), available from 8.10
-        $id = $args->arguments['id']
-            ?? $args->arguments[$model->getKeyName()]
-            ?? null;
+        $existingModel = null;
 
-        if ($id !== null) {
+        if (! empty($this->identifyingColumns)) {
```

Comment:
```suggestion
        if (isset($this->identifyingColumns)) {
```

The directive should implement `manipulateArgDefinition` and `manipulateFieldDefinition` to validate the given array is not empty and throw a `DefinitionException`.